ComponentKit/Core/Scope/CKDetectComponentScopeCollisions.mm

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKDetectComponentScopeCollisions.h"
+
+#import <queue>
+
+#import "CKAssert.h"
+#import "CKComponentInternal.h"
+
+void CKDetectComponentScopeCollisions(const CKComponentLayout &layout)
+{
+#if DEBUG
```

Comment:
@eczarny well in open source it's just NS_ASSERTIONS_ENABLED, and we have a private internal version at FB that you'd tie to the appropriate macro